ESPixelStick/src/service/DisplayOLED.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+* DisplayOLED.cpp
+*
+* Project: ESPixelStick - An ESP8266 / ESP32 and E1.31 based pixel driver
+* Copyright (c) 2018, 2024 Shelby Merrick
+* http://www.forkineye.com
+*
+*  This program is provided free for you to use in any way that you wish,
+*  subject to the laws and regulations where you are using it.  Due diligence
+*  is strongly suggested before using this code.  Please give credit where due.
+*
+*  The Author makes no warranty of any kind, express or implied, with regard
+*  to this program or the documentation contained in this document.  The
+*  Author shall not be liable in any event for incidental or consequential
+*  damages in connection with, or arising out of, the furnishing, performance
+*  or use of these programs.
+*
+*/
+
+#include "../ESPixelStick.h"
+#ifdef USE_OLED
+
+// OLED - Standard 128x32 i2c
+#include <SPI.h>
+#include <Wire.h>
+#include <Adafruit_GFX.h>
+#include <Adafruit_SSD1306.h>
+#include "DisplayOLED.h"
+#define SCREEN_WIDTH 128    // OLED display width, in pixels
+#define SCREEN_HEIGHT 32    // OLED display height, in pixels
+#define OLED_RESET -1       // Reset pin # (or -1 if sharing Arduino reset pin)
+#define SCREEN_ADDRESS 0x3C ///< See datasheet for Address; 0x3D for 128x64, 0x3C for 128x32
+Adafruit_SSD1306 display(SCREEN_WIDTH, SCREEN_HEIGHT, &Wire, OLED_RESET);
+
+void c_OLED::Begin ()
+{
+    // DEBUG_START;
+
+    // DEBUG_V(String("OLED Enabled: ") + String("WIDTH: " + SCREEN_WIDTH + " HEIGHT: " + SCREEN_HEIGHT + "ADDRESS: " +SCREEN_ADDRESS);
+   display.begin(SSD1306_SWITCHCAPVCC, SCREEN_ADDRESS);
```

Comment:
Which GPIOs are you using? There is no tie in with configuration of GPIOs and the display must use some pins someplace. These pins should be configurable. SPI and I2C pins are NOT constants in this system.

Reply:
Uses the pins defined in the HW platform (SDA SCL). A few of these changes could be accomplished with another HW variant. Maybe we start there ? I was trying to make it more broadly available to all of the hardware. Maybe the user ini?